src/vips-emscripten.cpp

```diff
@@ -29,6 +29,15 @@ using vips::TargetCustom;
 EM_JS(bool, is_node, (), { return ENVIRONMENT_IS_NODE; });
 #endif
 
+EM_JS(void, mark_all_threads_as_unused, (), {
+    if (!ENVIRONMENT_IS_NODE) {
+        return;
+    }
+    for (let worker of PThread.runningWorkers) {
```

Comment:
Btw, in theory we might want to mark `unusedWorkers` as `unref` here too until the referenced PR is merged, but the tests seem to pass fine regardless?

I'm not sure where the extra `+3` threads you're adding to the pool are used, would be worth manually testing in a scenario where they come into play just to be sure it still works.

Reply:
The fixed threadpool of libvips is started during `vips_init()`, including the `+3` threads specified in `PTHREAD_POOL_SIZE`, so all threads are probably in `PThread.runningWorkers` when this is reached.

FWIW, I just implemented the asynchronous queue implementation with commit https://github.com/kleisauke/libvips/commit/392831e4cf2823df86b2fb63b9e6ade385dbc172, which is currently being tested in CI with https://github.com/kleisauke/wasm-vips/commit/f9505ac1066d72b52771285e01f98ceb7878c916. This last commit clarifies why this `+3` is needed.